colossus/src/main/scala/colossus/protocols/http/HttpRequestParser.scala

```diff
@@ -25,13 +25,152 @@ object HttpRequestParser {
     } 
   }
 
-  protected def httpHead = firstline ~ headers >> {case method ~ path ~ version ~ headers => 
-    HttpHead(HttpMethod(method), path, HttpVersion(version), headers)
-  }
+  protected def httpHead = new HttpHeadParser
   
-  protected def firstline  = stringUntil(' ', minSize = Some(1), allowWhiteSpace = false) ~ 
-    stringUntil(' ', minSize = Some(1), allowWhiteSpace = false) ~ 
-    stringUntil('\r', minSize = Some(1), allowWhiteSpace = false) <~ byte
+}
+
+case class HeadResult(head: HttpHead, contentLength: Option[Int], transferEncoding: Option[String] )
```

Comment:
If HttpHead had built in lazy vals for fetching contentLength/transferEncoding, would this class be necessary?